javascript/code-challenges/graph-breadth-first/__tests__/graph-breadth-first.test.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const { Graph, Node } = require('../graph-breadth-first');
+
+describe('Graph', () => {
+  let graph;
+
+  beforeEach(() => {
+    graph = new Graph();
+
+    // Create nodes
+    const nodeA = new Node('A');
+    const nodeB = new Node('B');
+    const nodeC = new Node('C');
+    const nodeD = new Node('D');
+    const nodeE = new Node('E');
+    const nodeF = new Node('F');
+
+    // Connect nodes
+    nodeA.addAdjacentNode(nodeB);
+    nodeA.addAdjacentNode(nodeC);
+    nodeB.addAdjacentNode(nodeD);
+    nodeB.addAdjacentNode(nodeE);
+    nodeC.addAdjacentNode(nodeF);
```

Comment:
This is just a tree. We should also be testing to make sure it works if the graph has a cycle.

```js
nodeD.addAdjacentNode(nodeE);